Assets/Scripts/AudioManager.cs

```diff
@@ -0,0 +1,90 @@
+﻿using UnityEngine.Audio; 
+using UnityEngine;
+using System;
+
+public class AudioManager : MonoBehaviour
+{
+    public static AudioManager instance { get; private set; }
+    public Sound[] musicTracks;
+    public Sound[] sfxTracks;
+    
+
+    private void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+            return; 
+        }
+
+        foreach(Sound music in musicTracks)
+        {
+            music.source = gameObject.AddComponent<AudioSource>(); 
+            music.source.clip = music.audioClip;
+            music.source.loop = music.loop;
+            music.source.volume = music.volume;
+            music.source.pitch = music.pitch; 
+        }
+
+        foreach(Sound sfx in sfxTracks)
+        {
+            sfx.source = gameObject.AddComponent<AudioSource>();
+            sfx.source.clip = sfx.audioClip;
+            sfx.source.loop = sfx.loop;
+            sfx.source.volume = sfx.volume;
+            sfx.source.pitch = sfx.pitch;
+        }
```

Comment:
The double foreach loops here should be fine with regards to performance considering that we will not have many tracks – it should be very quick when our game starts up. Don't think there's reason to worry.

Reply:
SFX clips can be played using the Unity, AudioSource.PlayOneShot();

See this for reference: https://docs.unity3d.com/ScriptReference/AudioSource.PlayOneShot.html

I suggest you create one special Audiosource that handles all sfx, and use PlayOneShot to play them through there.